### PR TITLE
[DT-900] Get search working again

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DT-900

### Summary

Fixes the search bar, and ensure that it works properly with any combination of filters and when clearing the search.

To do this the `searchRef` was removed and the search input is treated as if it was any other filter. The only difference is that the input changes must also be reflected to the user, so we need to set the state when the user types or clears the filters.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/data_search/DatasetSearchTable.jsx [hunk 1]
```diff
@@ -76,12 +77,12 @@ export const DatasetSearchTable = (props) => {
...
 
-    // do not apply search modifier if there is no searchTerm
-    if (searchTerm !== '') {
+    // do not apply search modifier if there is no search term
+    if (filters.search.length > 0) {
       const searchModifier = [
         {
           'multi_match': {
-            'query': searchTerm,
+            'query': filters.search[filters.search.length - 1],
```
Comment (cinyecai) [inline src/components/data_search/DatasetSearchTable.jsx:85]: Why this change if searchTerm is being set in filterHandler?

Comment (fboulnois) [reply src/components/data_search/DatasetSearchTable.jsx:85]: For reasons I don't completely understand `searchTerm` at this line reflects the previous `searchTerm` state (for example if your current input was `hello`, `searchTerm` would be `hell`)

Comment (cinyecai) [reply src/components/data_search/DatasetSearchTable.jsx:85]: I'm wondering if something similar could be done everywhere searchTerm is used to eliminate the need for that variable at all - not sure if that would work right though

**Review by rushtong — APPROVED** [review]
👍🏽

**Review by snf2ye — APPROVED** [review]
